core/services/ping/pingutils.py

```diff
@@ -36,12 +39,25 @@ class PingDeviceDescriptor:
     firmware_version_minor: int
     firmware_version_patch: int
     port: SysFS
+    ethernet_discovery_info: Optional[str]
+    driver: Optional["PingDriver"]  # type: ignore
+
+    def get_hw_or_eth_info(self) -> str:
+        if self.port:
+            return str(self.port.device_path)
+        return str(self.ethernet_discovery_info)
 
     def __hash__(self) -> int:
-        return hash(self.port.hwid)
+        return hash(self.get_hw_or_eth_info())
```

Comment:
`ethernet_discovery_info` can be an empty string, in that case the function return value, that is used in around this PR with `==` with others variables and this hash function will give will have some problems, we should avoid or take care of that somehow, maybe moving `get_hw_or_eth_info` to an optional.

Reply:
ethernet_discovery_info should never be empty when `port is None`. I added a new commit for validating it